crates/rpc/src/lib.rs

```diff
@@ -18,15 +19,14 @@ impl BeerusRpc {
         Self { beerus }
     }
 
-    pub async fn run(
-        self,
-    ) -> Result<(SocketAddr, ServerHandle), BeerusRpcError> {
+    pub async fn run(self) -> eyre::Result<(SocketAddr, ServerHandle)> {
         let remote_spec_version =
             self.beerus.starknet_client.spec_version().await?;
 
         if remote_spec_version != SPEC_VERION {
-            return Err(BeerusRpcError::UnexpectedSpecVersion(
-                remote_spec_version,
+            return Err(eyre!(
```

Comment:
it would probably make message more informative if we provide what the expected version was